### ISSUE TITLE
<slot> cannot have directives

### ISSUE THREAD

**Issue by GeorgeTailor** [sveltejs/eslint-plugin-svelte3#6]
This is true when compiling standard Svelte components, but when the code is compiled to web-components `<slot>` can have directives and can be bound with `bind:this={slot}`.

Tested with `"svelte": "3.0.0-beta.20"` and `"eslint-plugin-svelte3@0.4.5"`.

**Comment by Conduitry** [comment]
:+1: There's not a way for the plugin to know what your build setup is or how you're going to be compiling the components, so I'll add another configurable setting that can be specified in your eslintrc that indicates whether custom element component validation or regular validation should be used.

**Comment by Conduitry** [comment]
Actually, if you tell the plugin that you are compiling with `customElement: true`, there's then also no way for it to know if you want to compile with a `tag` name passed in the compiler options, and it would always require one to be passed in `<svelte:options>`. I think the better and more general way to handle all of this is a new eslintrc setting that just lets you control the options that are passed to the compiler.

**Comment by GeorgeTailor** [comment]
Ok, cool, I guess this eslint option is not present yet, right ?

**Comment by Conduitry** [comment]
Right, I'm going to take a look at it soon though.

**Comment by Conduitry** [comment]
Just published v0.4.6 with a new `svelte3/compiler-options` eslintrc setting.

**Comment by quantuminformation** [comment]
How would I get round this limitation:

![image](https://user-images.githubusercontent.com/216566/125165734-91a94400-e198-11eb-85d6-9e7e5a3c9045.png)


```js
<script lang="ts">
	import { onMount } from 'svelte';
	import FadeInElement from '../../lib/molecules /FadeInElement.svelte';

	let observer;
	let element;
	export let once = false;

	function fadeInSection (element) {
		const callback = (entries) => {
			entries.forEach((entry) => {
				if (entry.isIntersecting) {
					element.classList.add('is-visible');
				}
			});
		};
		let options = {
			rootMargin: '0px',
			threshold: 0.5
		};

		observer = new IntersectionObserver(callback, options);

		observer.observe(element);
	}

	let section;

	onMount(() => {
		fadeInSection(section);

		return () => {
			debugger
			if (observer) observer.disconnect();
		};
	});
</script>


<slot bind:this={element}></slot>
```
